web-ui/src/app/components/step/parse/parse.component.ts

```diff
@@ -1,28 +1,29 @@
-import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
-import { AlpinoService } from '../../../services/_index';
+import { Component, Input, EventEmitter, Output } from '@angular/core';
+import { StepComponent } from '../step.component';
 
 @Component({
     selector: 'grt-parse',
     templateUrl: './parse.component.html',
     styleUrls: ['./parse.component.scss']
 })
-export class ParseComponent implements OnChanges {
-    @Input()
-    public xml: string;
-
-    @Input('sentence')
-    public sentence: string;
+export class ParseComponent extends StepComponent {
+    @Input() public loading = true;
+    @Input() public sentence: string;
+    @Input() public xml: string;
+    @Output() public changeXml = new EventEmitter<string>();
 
     public display = 'inline';
+    public validationMessage?: string;
 
-    public url: string
+    constructor() {
+        super();
+    }
 
-    constructor(private alpinoService: AlpinoService) {
+    public getValidationMessage(): string | void {
+        this.validationMessage = `Error parsing sentence: ${this.sentence}`; // we display our own error
```

Comment:
Ik krijg de error alleen te zien als ik hier de message return en dan ook alleen wanneer ik door probeer te klikken naar de volgende step. Na het refreshen van de pagina op deze stap mag ik altijd doorklikken naar de volgende stap, of er nou iets mis gaat of niet.

Reply:
Dit heb ik aangepakt met https://github.com/UUDigitalHumanitieslab/gretel/pull/203 ik heb daarvoor toch een begin gemaakt met een `StateService` omdat het anders wel echt te hackig werd.